controllers/stargate_controller.go

```diff
@@ -122,88 +122,120 @@ func (r *StargateReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
+			replicas += actualDeployment.Status.Replicas
+			readyReplicas += actualDeployment.Status.ReadyReplicas
+			updatedReplicas += actualDeployment.Status.UpdatedReplicas
+			availableReplicas += actualDeployment.Status.AvailableReplicas
 		}
 	}
 
-	// Check if the deployment needs to be updated
-	desiredDeploymentHash := desiredDeployment.Annotations[api.ResourceHashAnnotation]
-	if actualDeploymentHash, found := actualDeployment.Annotations[api.ResourceHashAnnotation]; !found || actualDeploymentHash != desiredDeploymentHash {
-		logger.Info("Updating Stargate Deployment", "Deployment", deploymentKey)
-		resourceVersion := actualDeployment.GetResourceVersion()
-		desiredDeployment.DeepCopyInto(actualDeployment)
-		actualDeployment.SetResourceVersion(resourceVersion)
+	for _, desiredDeployment := range desiredDeployments {
+		// Deployment does not exist yet: create a new one
+		deploymentKey := client.ObjectKey{Namespace: req.Namespace, Name: desiredDeployment.Name}
+		logger.Info("Stargate Deployment not found, creating a new one", "Deployment", deploymentKey)
 		// Set Stargate instance as the owner and controller
-		if err := ctrl.SetControllerReference(stargate, actualDeployment, r.Scheme); err != nil {
-			logger.Error(err, "Failed to set controller reference on updated Stargate Deployment", "Deployment", deploymentKey)
-			return ctrl.Result{}, err
-		} else if err := r.Update(ctx, actualDeployment); err != nil {
-			logger.Error(err, "Failed to update Stargate Deployment", "Deployment", deploymentKey)
+		if err := ctrl.SetControllerReference(stargate, &desiredDeployment, r.Scheme); err != nil {
+			logger.Error(err, "Failed to set controller reference on new Stargate Deployment", "Deployment", deploymentKey)
 			return ctrl.Result{}, err
+		} else if err := r.Create(ctx, &desiredDeployment); err != nil {
+			if errors.IsAlreadyExists(err) {
+				// the read from the local cache didn't catch that the resource was created
+				// already; simply requeue until the cache is up-to-date
+				return ctrl.Result{Requeue: true}, nil
+			} else {
+				logger.Error(err, "Failed to create new Stargate Deployment", "Deployment", deploymentKey)
+				return ctrl.Result{}, err
+			}
 		} else {
-			logger.Info("Stargate Deployment updated successfully", "Deployment", deploymentKey)
+			logger.Info("Stargate Deployment created successfully", "Deployment", deploymentKey)
 			return ctrl.Result{RequeueAfter: time.Minute}, nil
 		}
 	}
 
 	// Update status to reflect deployment status
-	if stargate.Status.Replicas != actualDeployment.Status.Replicas ||
-		stargate.Status.ReadyReplicas != actualDeployment.Status.ReadyReplicas ||
-		stargate.Status.UpdatedReplicas != actualDeployment.Status.UpdatedReplicas ||
-		stargate.Status.AvailableReplicas != actualDeployment.Status.AvailableReplicas {
-		ratio := fmt.Sprintf("%v/%v", actualDeployment.Status.ReadyReplicas, stargate.Spec.Size)
+	if stargate.Status.Replicas != replicas ||
```

Comment:
Now that a Stargate can have multiple Deployments, should the status be changed in some way to reflect that?

I haven't formed an opinion yet but wanted to go ahead and raise the question.

Reply:
I was asking myself that question too. I suggest that we take care of it in a follow-up PR, as I would like to revisit the whole logic for updating the Stargate status, and adopt an approach similar to the one you did for `K8ssandraClusterReconciler`.

Reply:
Sounds good. I should mention though that I am not sure I am happy with what I did in `K8ssandraClusterReconciler` 😬